src/Autoloader.php

```diff
@@ -0,0 +1,39 @@
+<?php
+/**
+ * @see       https://github.com/laminas/laminas-zendframework-bridge for the canonical source repository
+ * @copyright Copyright (c) 2019 Laminas Foundation (https://getlaminas.org)
+ * @license   https://github.com/laminas/laminas-zendframework-bridge/blob/master/LICENSE.md New BSD License
+ */
+
+namespace Laminas\ZendFrameworkBridge;
+
+/**
+ * Alias legacy Zend Framework project classes/interfaces/traits to Laminas equivalents.
+ */
+class Autoloader
+{
+    /**
+     * This autoloader is _append_, so a mix of legacy and Laminas classes can be used.
+     */
+    public static function load()
+    {
+        $classes = RewriteRules::namespaceRewrite();
+        spl_autoload_register(static function ($class) use ($classes) {
+            $segments = explode('\\', $class);
+
+            $i = 0;
+            $check = '';
+
+            while (isset($classes[$check . $segments[$i] . '\\'])) {
+                $check .= $segments[$i] . '\\';
+                ++$i;
+            }
```

Comment:
This seems really convoluted, honestly. Why not just:

```php
foreach ($class as $legacyPrefix => $aliasPrefix) {
    if (strpos($class, $legacyPrefix) === 0) {
        $alias = str_replace($legacyPrefix, $aliasPrefix, $class);
        class_alias($alias, $class);
        return;
    }
}
```

Reply:
It was exactly like that before. Now we are using segments, so we can return quicker - in 1 or two steps. With foreach we need to check it in the worst case scenario 12 times...

Reply:
Yep - I was misreading it previously.

Maybe put in a comment about how it works, so others who come to it later know why it's written the way it is.

Reply:
Comment is added now.